Terminal.Gui/Windows/AllowedType.cs

```diff
@@ -58,12 +66,21 @@ public override string ToString ()
 			return $"{this.Description} ({string.Join (";", this.Extensions.Select (e => '*' + e).ToArray ())})";
 		}
 
-		internal bool Matches (string extension, bool strict)
+		/// <inheritdoc/>
+		public bool IsAllowed(string path)
 		{
-			if (this.IsAny) {
-				return !strict;
+			if (this == Any) {
+				return true;
 			}
 
+			var extension = Path.GetExtension (path);
+
+			// There is a requirement to have a particular extension and we have none
+			if (string.IsNullOrEmpty (extension)) {
+				return false;
+			}
```

Comment:
And in the case where we want to open or save without extension? What the behavior?

Reply:
As a library user you can just implement `IAllowedType` interface yourself.  For example:

```csharp
public class RequireNoExtension : IAllowedType {
		public bool IsAllowed (string path)
		{
			return string.IsNullOrEmpty (Path.GetExtension (path));
		}
		public override string ToString ()
		{
			return "Must have no extension";
		}
}
```

Reply:
Ah ok injection here. Good work,, thanks.

Reply:
I've been thinking about the 'All Types' option.  This is currently implemented via  `AllowedTypesIsStrict` but I'm debating changing that.

![image](https://user-images.githubusercontent.com/31306100/226129756-e5f548ca-6266-423a-a3a1-ab85459ff5c1.png)

### Option 1 - AllowedTypesIsStrict
This is the way it currently is

```csharp
// Must be csv
fd.AllowedTypes = new List<IAllowedType> { new AllowedType("Delimited",".csv",".tsv") };
fd.AllowedTypesIsStrict = true;

// Can be csv or any other
fd.AllowedTypes = new List<IAllowedType> { new AllowedType ("Delimited", ".csv", ".tsv") };
```
### Option 2 - Just make the 'Any' option be another `IAllowedType` the user can add

In this case we would scrap the 'strict' flag and the user would just add the 'any' option explicitly.

```csharp
// Must be csv
fd.AllowedTypes = new List<IAllowedType> { new AllowedType ("Delimited", ".csv", ".tsv") };

// Can be csv or any
fd.AllowedTypes = new List<IAllowedType> { 
    new AllowedType ("Delimited", ".csv", ".tsv") ,
    new AllowedTypeAny()
};
```

Reply:
The option 2, no doubt. Why the user is bother creating additional extension and only going to use the first if `AllowedTypesIsStrict` is true.